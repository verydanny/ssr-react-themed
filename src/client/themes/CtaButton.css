@value ( RentBlue ) from 'ui/shared/themes/colors/colors.css';

@value ( BasicShadow, emailIcon, phoneIcon ) from 'ui/shared/themes/variables/variables.css';

.CtaButton {
  display: flex;
  align-items: center;
  background: RentBlue;
  box-shadow: BasicShadow;
  border-radius: 4px;
  background-size: 18px;
  background-repeat: no-repeat;
  background-position: 10px center;
  padding: 0 0 0 35px;
  font-size: 11px;
  color: #fff;
  text-decoration:none;
}

.CtaButton-email {
  background-image: emailIcon;
  border: 0px;
}


.CtaButton-phone {
  background-image: phoneIcon;
}

.CtaButton-phone-list,
.CtaButton-email-list,
.CtaButton-email-list-center {
  height: 25px;
}

.CtaButton-phone-map,
.CtaButton-email-map,
.CtaButton-email-map-center {
  width: 96px;
  height: 28px;
  box-sizing: content-box;
}

.CtaButton-phone-list {
  margin-bottom: 9px;
}
