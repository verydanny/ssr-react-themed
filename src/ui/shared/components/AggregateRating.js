import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed'
import { RatingBar } from '@rentpath/react-ui-core'

const roundedRating = rating => {
  if (!rating) return rating
  return Math.round(rating * 2.0) / 2.0
}

const Ratings = ({
  theme,
  listing,
  label,
  ...props
}) => {
  if (!listing.numRatings) return null

  const avgRating = roundedRating(listing.avgOverallRating)
  const effectiveLabel = label || `${listing.numRatings}`

  return (
    <RatingBar
      className={theme.AggregateRating}
      data-tid="rating"
      score={avgRating}
      label={effectiveLabel}
      {...props}
    />
  )
}

Ratings.propTypes = {
  theme: PropTypes.object,
  listing: PropTypes.shape({
    avgOverallRating: PropTypes.number,
    numRatings: PropTypes.number,
  }).isRequired,
  label: PropTypes.string,
}

Ratings.defaultProps = {
  theme: {},
}

export default themed(/^(AggregateRating)/, { pure: true })(Ratings)
