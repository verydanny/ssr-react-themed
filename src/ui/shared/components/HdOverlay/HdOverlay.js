import React from 'react'
import themed from 'react-themed'
import PropTypes from 'prop-types'

const HdOverlay = ({ theme }) => (
  <div
    data-tid="3dTour"
    className={theme.HdOverlay}
  >
    <span>3D</span> Tour
  </div>
)

HdOverlay.propTypes = {
  theme: PropTypes.object,
}

HdOverlay.defaultProps = {
  theme: {},
}

export default themed(/^HdOverlay/)(HdOverlay)
