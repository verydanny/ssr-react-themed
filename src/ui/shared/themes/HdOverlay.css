.HdOverlay {
  position: absolute;
  background: rgba(0, 0, 0, 0.4);
  color: #fff;
  font-size: 13px;
  line-height: 23px;
  width: 57px;
  height: 23px;
  text-align: center;
  white-space: nowrap;
  right: 238px;
  bottom: 0;
}

.HdOverlay span {
  font-weight: bold;
}
