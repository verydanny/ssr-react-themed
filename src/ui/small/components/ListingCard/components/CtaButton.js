import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed'
import cn from 'classnames'

const CtaButton = ({
  contactType,
  link,
  tags,
  onClick,
  variation,
  theme,
  children,
  nodeType: NodeType,
  ...others
}) => (
  <NodeType
    className={cn(
      theme.CtaButton,
      theme[`CtaButton-${contactType}`],
      variation && theme[`CtaButton-${contactType}-${variation}`],
    )}
    href={link}
    onClick={onClick}
    {...tags}
    {...others}
  >
    {children}
  </NodeType>
)

CtaButton.propTypes = {
  contactType: PropTypes.string.isRequired,
  link: PropTypes.string,
  tags: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  variation: PropTypes.string,
  theme: PropTypes.object,
  children: PropTypes.any,
  nodeType: PropTypes.string,
}

CtaButton.defaultProps = {
  theme: {},
  nodeType: 'button',
}

export default themed(/^CtaButton/)(CtaButton)
