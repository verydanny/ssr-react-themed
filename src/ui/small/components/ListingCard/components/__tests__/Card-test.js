import React from 'react'
import { shallow } from 'enzyme'
import AggregateRating from 'ui/shared/components/AggregateRating'
import HdOverlay from 'ui/shared/components/HdOverlay'
import CardThemed from '../Card'
import CtaButton from '../CtaButton'

const Card = CardThemed.WrappedComponent

describe('ListingCard/Card', () => {
  const baseProps = {
    theme: {},
    viewType: 'list',
    listing: {
      listingId: '1',
      isActive: true,
      mPhone: '123-456-7890',
      noVacancy: false,
      isPaid: true,
    },
  }
  const genericSpotlightClass = 'sponsored-class-for-all'
  const ctaView = 'sponsored-class-for-list'
  const mapViewProps = {
    theme: {
      CardInfo_sponsoredLabel: genericSpotlightClass,
    },
    viewType: 'map',
    listing: {
      listingId: 1,
    },
    selectedListing: '1',
  }

  const listViewProps = {
    theme: {
      ctaView,
      CardInfo_sponsoredLabel: genericSpotlightClass,
    },
    viewType: 'list',
    isListView: true,
    listing: {
      listingId: 1,
    },
    selectedListing: '1',
  }

  const propsForSpotlight = {
    ...listViewProps,
    listing: {
      ...listViewProps.listing,
      isCurrentSpotlight: true,
    },
  }

  describe('3D Tour card', () => {
    describe('listing with 3d tour', () => {
      const tourProp = { hasHdTour: true }
      const hdListing = { ...baseProps, ...tourProp }
      it('has 3D Tour label', () => {
        const wrapper = shallow(
          <Card
            listing={hdListing}
          />
        )
        expect(wrapper.find(HdOverlay).length).toEqual(1)
      })
    })
    describe('listing without 3d tour', () => {
      it('does not have 3D Tour label', () => {
        const wrapper = shallow(
          <Card
            listing={baseProps}
          />
        )
        expect(wrapper.find(HdOverlay).length).toEqual(0)
      })
    })
  })

  describe('when listing is inactive', () => {
    it('does not show any CTAs', () => {
      const wrapper = shallow(
        <Card {...baseProps} listing={{ ...baseProps.listing, isActive: false }} />
      )
      expect(wrapper.find(CtaButton)).toHaveLength(0)
    })
  })

  it('shows CTAs when the listing is active', () => {
    const wrapper = shallow(<Card {...baseProps} />)
    expect(wrapper.find(CtaButton)).toHaveLength(2)
  })

  it('renders the aggregate ratings', () => {
    const wrapper = shallow(<Card {...baseProps} />)
    expect(wrapper.find(AggregateRating)).toHaveLength(1)
  })

  describe('spotlight', () => {
    describe('when on list view', () => {
      describe('when listing is current spotlight', () => {
        it('renders a spotlight with ctaView class', () => {
          const wrapper = shallow(<Card {...propsForSpotlight} />)
          expect(wrapper.find(`.${ctaView}`)).toHaveLength(1)
        })

        it('renders a spotlight with CardInfo_sponsoredLabel', () => {
          const wrapper = shallow(<Card {...propsForSpotlight} />)
          expect(wrapper.find(`.${genericSpotlightClass}`)).toHaveLength(1)
        })
      })

      describe('when listing is not current spotlight', () => {
        it('does not render a spotlight with ctaView class', () => {
          const wrapper = shallow(<Card {...listViewProps} />)
          expect(wrapper.find(`.${ctaView}`)).toHaveLength(0)
        })

        it('does not render a spotlight with CardInfo_sponsoredLabel', () => {
          const wrapper = shallow(<Card {...listViewProps} />)
          expect(wrapper.find(`.${genericSpotlightClass}`)).toHaveLength(0)
        })
      })
    })

    describe('when on map view', () => {
      describe('when listing is current spotlight', () => {
        it('renders a spotlight', () => {
          const wrapper = shallow(<Card {...propsForSpotlight} />)
          expect(wrapper.find(`.${genericSpotlightClass}`)).toHaveLength(1)
        })
      })

      describe('when listing is not current spotlight', () => {
        it('does not render a spotlight', () => {
          const wrapper = shallow(<Card {...mapViewProps} />)
          expect(wrapper.find(`.${genericSpotlightClass}`)).toHaveLength(0)
        })
      })

      describe('snapshot', () => {
        it('renders correctly', () => {
          const wrapper = shallow(<Card {...mapViewProps} />)
          expect(wrapper).toMatchSnapshot()
        })
        it('renders spotlight correctly', () => {
          const wrapper = shallow(<Card {...propsForSpotlight} />)
          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })
})
