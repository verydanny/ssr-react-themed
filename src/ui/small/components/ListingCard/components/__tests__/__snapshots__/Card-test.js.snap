// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCard/Card spotlight when on map view snapshot renders correctly 1`] = `
<div
  className=""
  data-tag_item="padding_box"
  data-tag_listing_id={1}
  data-tag_section={null}
  data-tid="listing-section"
  id=""
  onClick={[Function]}
  role="presentation"
>
  <span
    itemScope={true}
    itemType="http://schema.org/ApartmentComplex"
  >
    <div>
      <div
        className=""
      />
      <div>
        <div
          className=""
        >
          <Themed(Ratings)
            listing={
              Object {
                "listingId": 1,
              }
            }
          />
        </div>
      </div>
    </div>
    <Connect(Themed(ImageGallery))
      dataTagItem="image"
    />
    <Connect(Themed(ListingInfo))
      isActiveCard={true}
      listingId={1}
      viewType="map"
    />
  </span>
  <Themed(FavoriteIcon)
    isFavorite={false}
    onClick={[Function]}
  />
</div>
`;

exports[`ListingCard/Card spotlight when on map view snapshot renders spotlight correctly 1`] = `
<div
  className="sponsored-class-for-list"
  data-tag_item="padding_box"
  data-tag_listing_id={1}
  data-tag_section="spotlight"
  data-tid="listing-section"
  id="list-1"
  onClick={[Function]}
  role="presentation"
>
  <span
    itemScope={true}
    itemType="http://schema.org/ApartmentComplex"
  >
    <Tagging
      spotlight_listing_ids={1}
    />
    <div>
      <div
        className=""
      />
      <div>
        <div
          className="undefined"
        >
          <Themed(Ratings)
            listing={
              Object {
                "isCurrentSpotlight": true,
                "listingId": 1,
              }
            }
          />
          <span
            className="sponsored-class-for-all"
            data-tid="sponsored-listing-banner"
          >
            Sponsored
          </span>
        </div>
      </div>
    </div>
    <Connect(Themed(ImageGallery))
      dataTagItem="image"
    />
    <Connect(Themed(ListingInfo))
      isActiveCard={true}
      listingId={1}
      viewType="list"
    />
  </span>
  <Themed(FavoriteIcon)
    isFavorite={false}
    onClick={[Function]}
  />
</div>
`;
